feature/tpm/swtpm_test.go

```diff
@@ -0,0 +1,441 @@
+	dataDir    string
+	deviceName string
+	devicePath string
+	pidFile    string
+	opts       *swtpmOptions
+	cmd        *exec.Cmd
+	t          testing.TB
+}
+
+func newSWTPM(t testing.TB, opts ...swtpmOption) *swtpm {
+	t.Helper()
+	options := &swtpmOptions{
+		version: "2.0",
+	}
+
+	for _, opt := range opts {
+		opt(options)
+	}
+
+	dataDir := t.TempDir()
+
+	suffix := make([]byte, 8)
+	if _, err := crand.Read(suffix); err != nil {
+		t.Fatalf("failed to generate random suffix: %v", err)
+	}
+
+	// use unique per-test vtpm device names to avoid conflicts
+	deviceName := fmt.Sprintf("vtpm-%d-%s", time.Now().UnixNano(), hex.EncodeToString(suffix))
+	devicePath := filepath.Join("/dev", deviceName)
+	pidFile := filepath.Join(dataDir, "swtpm.pid")
+
+	s := &swtpm{
+		dataDir:    dataDir,
+		deviceName: deviceName,
+		devicePath: devicePath,
+		pidFile:    pidFile,
+		opts:       options,
+		t:          t,
+	}
+	s.t.Logf("created swtpm with device %s, data dir %s", deviceName, dataDir)
+
+	if err := s.start(t.Context()); err != nil {
+		t.Fatalf("failed to start swtpm: %v", err)
+	}
+
+	t.Cleanup(func() {
+		s.stop()
+	})
+
+	return s
+}
+
+// runSetup initializes the TPM state using swtpm_setup
+func (s *swtpm) runSetup() error {
+	args := []string{
+		"--tpmstate", s.dataDir,
+	}
+
+	switch s.opts.version {
+	case "1.2":
+		args = append(args, "--tpm-1.2")
+	case "2.0":
+		args = append(args, "--tpm2")
+	}
+
+	cmd := exec.Command("swtpm_setup", args...)
+	s.t.Logf("running swtpm_setup with args: %v", args)
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("swtpm_setup failed: %w", err)
+	}
+
+	return nil
+}
+
+// start launches the swtpm process with the configured options
+func (s *swtpm) start(ctx context.Context) error {
+	// init state if requested
+	if s.opts.withSetup {
+		if err := s.runSetup(); err != nil {
+			return fmt.Errorf("swtpm_setup failed: %w", err)
+		}
+	}
+
+	args := []string{
+		"cuse",
+		"--tpmstate", fmt.Sprintf("dir=%s", s.dataDir),
+		"--name", s.deviceName,
+		"--pid", fmt.Sprintf("file=%s", s.pidFile),
+	}
+
+	switch s.opts.version {
+	case "1.2":
+	case "2.0":
+		args = append(args, "--tpm2")
+	default:
+		return fmt.Errorf("unsupported swtpm version: %s", s.opts.version)
+	}
+
+	// when using swtpm_setup, we need to tell swtpm to send TPM2_Startup(CLEAR)
+	if s.opts.withSetup {
+		args = append(args, "--flags", "startup-clear")
+	}
+
+	if s.opts.flags != nil {
+		args = append(args, s.opts.flags...)
+	}
+
+	cmd := exec.Command(swtpmBinary, args...)
+	cmd.Stdout = os.Stderr
+	cmd.Stderr = os.Stderr
+	if err := cmd.Start(); err != nil {
+		return fmt.Errorf("failed to start swtpm: %w", err)
+	}
+
+	s.cmd = cmd
+	if err := s.waitForDevice(); err != nil {
+		s.stop()
+		return fmt.Errorf("swtpm device not available: %w", err)
+	}
+
+	return nil
+}
+
+// waitForDevice waits for the swtpm character device to be created
+func (s *swtpm) waitForDevice() error {
+	timeout := time.After(1 * time.Second)
```

Comment:
nit: just to avoid flakiness on an overloaded CI host
```suggestion
	timeout := time.After(5 * time.Second)
```